src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -177,30 +177,62 @@ class EntityClient (requestContext: RequestContext) extends Actor {
+            HttpEntity(MediaTypes.`application/json`,calls.toJson.toString))
+    }
 
-    //STUB: return a batch of AddUpdateOperations
+    responseFuture onComplete {
+      case Success(response) =>
+        response.status match {
+          case NoContent =>
+            log.debug("OK response")
+            requestContext.complete(response)
+          case _ =>
+            // Bubble up all other unmarshallable responses
+            log.warning("Unanticipated response: " + response.status.defaultMessage)
+            requestContext.complete(response)
+        }
+      case Failure(error) =>
+        // Failure accessing service
+        log.error(error, "Service API call failed")
+        requestContext.failWith(
+          new RequestProcessingException(StatusCodes.InternalServerError, error.getMessage))
+    }
   }
 
-  def batchCallToRawls( calls: Any ) = {
-    //STUB: stitch the batch calls together and fire off the Rawls request.
-    //requestContext.complete(...)
+  val upsertAttrOperation = "op" -> AttributeString("AddUpdateAttribute")
+  val removeAttrOperation = "op" -> AttributeString("RemoveAttribute")
+  val addListMemberOperator = "op" -> AttributeString("AddListMember")
+
+  def setAttributesOnEntity(entityType: String, row: Seq[String], colInfo: Seq[(String,Option[String])]) = {
+    val ops = for { (value,(attributeName,refTypeOpt)) <- row.tail zip colInfo if refTypeOpt.isDefined || !value.isEmpty } yield {
+      val nameEntry = "attributeName" -> AttributeString(attributeName)
+      def valEntry( attr: Attribute ) = "addUpdateAttribute" -> attr
+      refTypeOpt match {
+        case Some(refType) => Map(upsertAttrOperation,nameEntry,valEntry(AttributeReference(refType,value)))
+        case None => value match {
+          case "__DELETE__" => Map(removeAttrOperation,nameEntry)
+          case _ => Map(upsertAttrOperation,nameEntry,valEntry(AttributeString(value)))
+        }
+      }
+    }
+    EntityUpdateDefinition(row(0),entityType,ops)
   }
 
-  def addToCollectionTypeFromTSV( workspaceNamespace: String, workspaceName: String, entityType: String, memberType: String, membersAttributeName: String, tsv: TSVLoadFile ) = {
-    //group together updates to distinct entities, construct the list of entities to add references to, and poke it into whatever the "members" field is called.
-    val tsvFieldName = memberType + "_id"
-    tsv.tsvData.groupBy( _(tsv.firstColumnHeader) ).map({
-      case (entityName, rows) => addToCollectionMembersAttribute( entityType, entityName, rows.map(row => (memberType, row(tsvFieldName))), membersAttributeName )
-    })
+  def addToCollectionTypeFromTSV( entityType: String, memberType: String, membersAttributeName: String, tsv: TSVLoadFile ) = {
+    // group together updates to distinct entities
+    tsv.tsvData groupBy(_(0)) map { case (entityName, rows) =>
+      val ops = rows map { row =>
+        val attrRef = AttributeReference(memberType,row(1))
```

Comment:
here and other places knowing what 0 and 1 are would be helpful, 0 in most places i get, first column, but val some_useful_variable_name = row(1) would be nice